### PR TITLE
User security

### PR DESCRIPTION
**Pull request by yassersh12**
This PR added the security part for user endpoints and introduced a get_current_active_admin method to enforce admin-only access.

### CODE REVIEW

**Review by Omarjabari007 — APPROVED** [review]

File: app/api/dependencies/auth.py [hunk 1]
```diff
@@ -95,3 +95,21 @@ async def get_current_active_user(
     except Exception as e:
         print(f"An error occurred while getting current active user: {e}")
         raise HTTPException(status_code=400, detail="Error getting current active user")
+
+
+async def get_current_active_admin(
+    current_user: Annotated[User, Depends(get_current_active_user)],
+):
+    try:
+        if not current_user.active:
```
Comment (FaresAbuIram) [inline app/api/dependencies/auth.py]: No need for this part, already implemented in line 92

File: app/api/routes/user.py [hunk 2]
```diff
@@ -10,18 +10,37 @@
...
+# this method for normal user
+@router.post("/register", response_model=UserResponse)
+def register_user(user: UserCreateRequest, db: Session = Depends(get_db)):
+    service = UserService(db)
+    user_dict = user.model_dump()
+    user_dict["is_admin"] = False
+    return service.create_user(UserCreateRequest(**user_dict))
+
+
+# this method only for admins to regiter users
```
Comment (FaresAbuIram) [inline app/api/routes/user.py:27]: I don't think we need this method while we have a change role endpoint

Comment (FaresAbuIram) [reply app/api/routes/user.py:27]: But we can keep it, just for your information

**Review by FaresAbuIram — CHANGES_REQUESTED** [review]
Well done 
Small changes

**Review by FaresAbuIram — APPROVED** [review]

**Pull request merged**